ampd/src/lib.rs

```diff
@@ -275,7 +281,24 @@ where
         };
 
         while (set.join_next().await).is_some() {}
-
-        res
+        state_rx.try_recv().change_context(Error::ReturnState)
```

Comment:
Now this run function wouldn't return back the error from the first terminated task, if any. I don't think this is right.